cpp/src/arrow/buffer.h

```diff
@@ -47,9 +47,25 @@ class MemoryPool;
 /// The following invariant is always true: Size < Capacity
 class ARROW_EXPORT Buffer {
  public:
+  /// \brief Construct from buffer and size without copying memory
+  ///
+  /// \param[in] data a memory buffer
+  /// \param[in] size buffer size
+  ///
+  /// \note The passed memory must be kept alive through some other means
```

Comment:
Would ASAN be able to detect this kind of lifetime error?

Reply:
I think so, and you would segfault if you tried accessing the memory that had already been freed